assets/src/stories-editor/components/video-block-edit-with-preview.js

```diff
@@ -0,0 +1,296 @@
+		if ( this.props.attributes.poster !== prevProps.attributes.poster ) {
+			this.videoPlayer.current.load();
+		}
+	}
+	toggleAttribute( attribute ) {
+		return ( newValue ) => {
+			this.props.setAttributes( { [ attribute ]: newValue } );
+		};
+	}
+
+	onSelectURL( newSrc ) {
+		const { attributes, setAttributes } = this.props;
+		const { src } = attributes;
+
+		// Set the block's src from the edit component's state, and switch off
+		// the editing UI.
+		if ( newSrc !== src ) {
+			// Omit the embed block logic, as that didn't seem to work.
+			setAttributes( { src: newSrc, id: undefined } );
+		}
+
+		this.setState( { editing: false } );
+	}
+
+	onSelectPoster( image ) {
+		const { setAttributes } = this.props;
+		setAttributes( { poster: image.url } );
+	}
+
+	onRemovePoster() {
+		const { setAttributes } = this.props;
+		setAttributes( { poster: '' } );
+
+		// Move focus back to the Media Upload button.
+		this.posterImageButton.current.focus();
+	}
+
+	onUploadError( message ) {
+		const { noticeOperations } = this.props;
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice( message );
+	}
+
+	render() {
+		const {
+			caption,
+			controls,
+			poster,
+			src,
+		} = this.props.attributes;
+		const {
+			className,
+			instanceId,
+			isSelected,
+			noticeUI,
+			setAttributes,
+		} = this.props;
+		const { editing } = this.state;
+		const switchToEditing = () => {
+			this.setState( { editing: true } );
+		};
+		const onSelectVideo = ( media ) => {
+			if ( ! media || ! media.url ) {
+				// in this case there was an error and we should continue in the editing state
+				// previous attributes should be removed because they may be temporary blob urls
+				setAttributes( { src: undefined, id: undefined } );
+				switchToEditing();
+				return;
+			}
+			// sets the block's attribute and updates the edit component from the
+			// selected media, then switches off the editing UI
+			setAttributes( { src: media.url, id: media.id } );
+			this.setState( { src: media.url, editing: false } );
+		};
+
+		if ( editing ) {
+			return (
+				<MediaPlaceholder
+					icon={ <BlockIcon icon={ icon } /> }
+					className={ className }
+					onSelect={ onSelectVideo }
+					onSelectURL={ this.onSelectURL }
+					accept="video/mp4"
+					allowedTypes={ ALLOWED_MEDIA_TYPES }
+					value={ this.props.attributes }
+					notices={ noticeUI }
+					onError={ this.onUploadError }
+				/>
+			);
+		}
+
+		const videoPosterDescription = `video-block__poster-image-description-${ instanceId }`;
+
+		return (
+			<>
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className="components-icon-button components-toolbar__control"
+							label={ __( 'Edit video' ) }
+							onClick={ switchToEditing }
+							icon="edit"
+						/>
+					</Toolbar>
+				</BlockControls>
+				<InspectorControls>
+					<PanelBody title={ __( 'Video Settings' ) }>
+						<ToggleControl
+							label={ __( 'Playback Controls' ) }
```

Comment:
Uhm, it doesn't seem like controls are actually displayed in AMP Stories? The runtime seems to hide them no matter what.

@westonruter Can you confirm this? Should we just remove this option?

Reply:
AFAIK this was already removed at some point.

Found also this: https://github.com/ampproject/amp-wp/pull/2255 where `controls` was set to `false` by default.

Reply:
Hmm so we should remove the `controls` setting, but keep the `loop` (default to true) and `mute` settings?

Even though `mute` doesn't seem to have any effect. When opening an AMP Story, sound is muted by default for the whole story (see upper right corner), and when I enable sound, the video is not muted.

Reply:
Yep, the `mute` control doesn't seem to have any effect anyway, looks like "Loop" is the only control that actually does anything.

Seems like we could probably just add the `loop` and leave the rest out for now.

Reply:
Added back the `loop` setting and removed the one for `controls`.

If `loop` is disabled, `controls` are still shown since one might want to re-start the video during editing.